compose/src/main/kotlin/io/github/kakaocup/compose/intercept/interaction/ComposeInteraction.kt

```diff
@@ -3,16 +3,31 @@ package io.github.kakaocup.compose.intercept.interaction
 import androidx.compose.ui.test.SemanticsNodeInteraction
 import io.github.kakaocup.compose.intercept.operation.ComposeAction
 import io.github.kakaocup.compose.intercept.operation.ComposeAssertion
+import io.github.kakaocup.compose.node.builder.NodeProvider
 
 class ComposeInteraction(
-    private val semanticsNodeInteraction: SemanticsNodeInteraction
+    val nodeProvider: NodeProvider,
 ) : Interaction<ComposeAssertion, ComposeAction> {
 
+    var semanticsNodeInteraction: SemanticsNodeInteraction = nodeProvider.provideSemanticsNodeInteraction()
+        private set
+
     override fun check(assertion: ComposeAssertion) {
-        assertion.invoke(semanticsNodeInteraction)
+        assertion.execute(semanticsNodeInteraction)
     }
 
     override fun perform(action: ComposeAction) {
-        action.invoke(semanticsNodeInteraction)
+        action.execute(semanticsNodeInteraction)
+    }
+
+    fun reFindNode() {
```

Comment:
This method is used in the Kaspresso interceptor to try finding the Node after exception. SemanticsNodeInteraction is not updating while the Tree is changing, that's why it's better to search the Node again using a previous Matcher.